src/SAWScript/Builtins.hs

```diff
@@ -203,11 +203,13 @@ readSBV path unintlst =
 -- support for latches in the 'AIGNetwork' SAWScript type.
 dsecPrint :: TypedTerm -> TypedTerm -> TopLevel ()
 dsecPrint t1 t2 = do
-  withSystemTempFile ".aig" $ \path1 _handle1 -> do
-  withSystemTempFile ".aig" $ \path2 _handle2 -> do
-  Prover.writeSAIGInferLatches path1 t1
-  Prover.writeSAIGInferLatches path2 t2
-  io $ callCommand (abcDsec path1 path2)
+  write_t1 <- Prover.writeSAIGInferLatches t1
```

Comment:
I think these AIG-related changes might be unrelated work that was caught in the first commit.

Reply:
This is actually related, although the reason is pretty subtle.  `TopLevel` no longer has a `MonadMask` instance, as continuation-based monads don't provide the necessary guarantees. However `withSystemTempFile` demands `MonadMask` in order to manage the temporary file lifetime. So, one solution is this, where we extract an IO action from the `TopLevel` and execute it later in an IO context.

This seems to be the only place that `MonadMask` instance was being used.